File: app/src/main/java/fr/frogdevelopment/nihongo/dico/data/rest/SentencesClient.java
package fr.frogdevelopment.nihongo.dico.data.rest;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface SentencesClient {

    @GET("sentences/search")
    Call<List<Sentence>> search(@Query("lang") String lang,
                                @Query("kanji") String kanji,
                                @Query("kana") String kana,
                                @Query("gloss") String gloss);
}

File: app/src/main/java/fr/frogdevelopment/nihongo/dico/data/rest/SentencesClient.kt
package fr.frogdevelopment.nihongo.dico.data.rest

import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query

interface SentencesClient {

    @GET("nihongo-dico-sentences/search")
    fun search(@Query("lang") lang: String,
               @Query("kanji") kanji: String?,
               @Query("kana") kana: String,
               @Query("gloss") gloss: String): Call<List<Sentence>?>
}